File: src/main/java/com/vjiki/music/dto/StoryResponse.java
package com.vjiki.music.dto;

import java.time.OffsetDateTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Instagram-style story response
 * Optimized for mobile story UI
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class StoryResponse {
    private UUID id;
    private UUID userId;
    private String userNickname;
    private String userAvatarUrl;
    private String imageUrl;
    private String previewUrl;
    private String storyType;
    private UUID songId;
    private String songTitle;
    private String songArtist;
    private String caption;
    private String location;
    private Integer viewsCount;
    private OffsetDateTime createdAt;
    private OffsetDateTime expiresAt;

    @JsonProperty("isExpired")
    private Boolean isExpired;
}

File: src/main/kotlin/com/vjiki/music/dto/StoryResponse.kt
package com.vjiki.music.dto

import java.time.OffsetDateTime
import java.util.UUID

/**
 * Instagram-style story response
 * Optimized for mobile story UI
 */
data class StoryResponse(
    val id: UUID,
    val userId: UUID,
    val userNickname: String?,
    val userAvatarUrl: String?,
    val imageUrl: String?,
    val previewUrl: String?,
    val storyType: String,
    val songId: UUID?,
    val songTitle: String?,
    val songArtist: String?,
    val caption: String?,
    val location: String?,
    val viewsCount: Int,
    val createdAt: OffsetDateTime?,
    val expiresAt: OffsetDateTime?,
    val isExpired: Boolean
)

